tesseract_command_language/test/joint_waypoint_test.cpp

```diff
@@ -0,0 +1,56 @@
+/**
+ * @file joint_waypoint_unit.cpp
+ * @brief
+ *
+ * @author Levi Armstrong
+ * @date January 28, 2021
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2021, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <gtest/gtest.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+#include <tesseract_command_language/joint_waypoint.h>
+
+using namespace tesseract_planning;
+
+TEST(TesseractCommandLanguageJointWaypointUnit, isToleranced)
+{
+  std::vector<std::string> joint_names = { "joint_1", "joint_2", "joint_3" };
+  Eigen::VectorXd joint_values = Eigen::VectorXd::Constant(3, 1);
+
+  JointWaypoint jw(joint_names, joint_values);
+  EXPECT_FALSE(jw.isToleranced());
+
+  jw.upper_tolerance = Eigen::VectorXd::Constant(3, 5);
+  jw.lower_tolerance = Eigen::VectorXd::Constant(3, -5);
```

Comment:
I think this technically violates your description of the tolerances:

>   /** Joint distance below waypoint that is allowed. **Each element should be <= 0** */
  Eigen::VectorXd lower_tolerance;

Reply:
I am not following how this violate the description; is there something I am missing? The lower tolerance is being populated with a vector of length 3 with a constant value of `-5`. All values in the vector should be `-5` which is `<=0`.

Reply:
Sorry, I misread this. Disregard my previous comment